File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/CSharp/Web/1033/AjaxControlExtender/ExtenderControl1.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Linq;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
$if$ ($targetframeworkversion$ >= 3.5)using System.Xml.Linq;
$endif$

namespace $safeprojectname$
{
    /// <summary>
    /// Summary description for ExtenderControl1
    /// </summary>
    [
        TargetControlType(typeof(Control))
    ]
    public class ExtenderControl1 : ExtenderControl
    {
	    public ExtenderControl1()
	    {
		    //
		    // TODO: Add constructor logic here
		    //
	    }
        protected override IEnumerable<ScriptDescriptor>
                GetScriptDescriptors(System.Web.UI.Control targetControl)
        {
            ScriptBehaviorDescriptor descriptor = new ScriptBehaviorDescriptor("$safeprojectname$.ClientBehavior1", targetControl.ClientID);
            yield return descriptor;
        }

        // Generate the script reference
        protected override IEnumerable<ScriptReference>
                GetScriptReferences()
        {
            yield return new ScriptReference("$safeprojectname$.ClientBehavior1.js", this.GetType().Assembly.FullName);
        }
    }
}

File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/VisualBasic/Web/Version2012/1033/AjaxControlExtender/ExtenderControl1.vb
<TargetControlType(GetType(Control))> _
Public Class ExtenderControl1
    Inherits ExtenderControl

    Protected Overrides Function GetScriptDescriptors(ByVal targetControl As System.Web.UI.Control) As IEnumerable(Of ScriptDescriptor)
        Dim descriptor As New ScriptBehaviorDescriptor("$safeprojectname$.ClientBehavior1", targetControl.ClientID)

        Return New List(Of ScriptDescriptor) From {descriptor}
    End Function

    ' Generate the script reference
    Protected Overrides Function GetScriptReferences() As IEnumerable(Of ScriptReference)
        Dim scriptRef As New ScriptReference("$safeprojectname$.ClientBehavior1.js", Me.GetType().Assembly.FullName)

        Return New List(Of ScriptReference) From {scriptRef}
    End Function
End Class
